src/redis-cli.c

```diff
@@ -727,7 +727,7 @@ static int cliSelect(void) {
 /* Select RESP3 mode if redis-cli was started with the -3 option.  */
 static int cliSwitchProto(void) {
     redisReply *reply;
-    if (config.resp3 == 0 || config.resp2 == 1) return REDIS_OK;
+    if (config.resp3 == 0 || config.resp2) return REDIS_OK;
```

Comment:
```suggestion
    if (config.resp3 == 0 || config.resp2 != 0) return REDIS_OK;
```
nit pick: i'd prefer to either explicitly compare both to 0, or use `if (!config.resp3 || config.resp2)`